docs/releases/6.2.md

```diff
@@ -81,4 +81,76 @@ WAGTAILFRONTENDCACHE = {
 
 ## Upgrade considerations - changes affecting Wagtail customisations
 
+### Changes to report views with the new Universal Listings UI
+
+The report views have been reimplemented to use the new Universal Listings UI, which introduces AJAX-based filtering and support for the `wagtail.admin.ui.tables` framework.
+As a result, a number of changes have been made to the `ReportView` and `PageReportView` classes, as well as their templates.
+
+If you have custom report views as documented in [](adding_reports), you will need to make the following changes.
+
+#### Change `title` to `page_title`
+
+The `title` attribute on the view class should be renamed to `page_title`:
+
+```diff
+ class UnpublishedChangesReportView(PageReportView):
+-    title = "Pages with unpublished changes"
++    page_title = "Pages with unpublished changes"
+```
+
+#### Set up the results-only view
+
+Set the `index_url_name` and `index_results_url_name` attributes on the view class:
+
+```diff
+ class UnpublishedChangesReportView(PageReportView):
++    index_url_name = "unpublished_changes_report"
++    index_results_url_name = "unpublished_changes_report_results"
+```
+
+and register the results-only view:
+
+```diff
+ @hooks.register("register_admin_urls")
+ def register_unpublished_changes_report_url():
+     return [
+         path("reports/unpublished-changes/", UnpublishedChangesReportView.as_view(), name="unpublished_changes_report"),
++        # Add a results-only view to add support for AJAX-based filtering
++        path("reports/unpublished-changes/results/", UnpublishedChangesReportView.as_view(results_only=True), name="unpublished_changes_report_results"),
+     ]
+```
+
+#### Adjust the templates
+
+If you are only extending the templates to add your own markup for the listing table, you need to change the `template_name` into `results_template_name` on the view class.
+
+```diff
+ class UnpublishedChangesReportView(PageReportView):
+-    template_name = "reports/unpublished_changes_report.html"
++    results_template_name = "reports/unpublished_changes_report_results.html"
+```
+
+After renaming the template (e.g. `reports/unpublished_changes_report.html` to `reports/unpublished_changes_report_results.html`), you will need to make a few changes to the template's contents:
+
+- The template should extend from `wagtailadmin/reports/base_page_report_results.html`.
+- The `listing` and `no_results` blocks should be renamed to `results` and `no_results_message`, respectively.
```

Comment:
It would be worth mentioning the change for non-page reports too - while these aren't fully documented, the "adding reports" page does briefly mention `wagtailadmin/reports/base_report.html`, so developers might reasonably have reports based on that and need to know how to handle them.

```suggestion
- The `listing` and `no_results` blocks should be renamed to `results` and `no_results_message`, respectively.
- For a non-page report extending from `wagtailadmin/reports/base_report.html`, this should be changed to `wagtailadmin/reports/base_report_results.html`. Existing templates will typically define a `results` block containing both the results listing and the "no results" message; these should now become separate blocks named `results` and `no_results_message`.
```

(or if you prefer, make this a separate pair of bullet points after the unpublished_changes_report_results.html code snippet...)

Reply:
Thanks! I've decided to separate the instructions and code snippets for non-page reports in ab8d721f92bdc744321cb0b81cd9f7a8721a512d. I also added a paragraph to note that having a `results_template_name` (and a results-only template) is still necessary if you want to customise the `template_name`.